### DISCUSSION TITLE
change multiple /sethome and /home uses to be consistent with permissions; also economy stuff

### DISCUSSION THREAD

**Discussion by Tavies** [EssentialsX/Essentials#3977] [Ideas and Feature Suggestions] [answered]
After elevating permissions to allow for more homes to be set, a player has continued access to that number of homes even when removing the permission (bug or feature?). It would be ideal to have a config option to hide the extra homes until the permission is restored.

An example: a default player sets 3 homes then gets promoted to vip for a month and sets the 2 additional homes for their total of 5 homes. After the vip permission expires,  the two extra homes are hidden from use until they regain the vip permission again.

Current behavior allows for that player to continue to set all 5 homes, though deleting any of the homes (eg: to rename) compresses the home list (deleting number 3 would shift 4 and 5 down a spot). Having the homes stay in the same slot and being able to manipulate them (delete 3 and have an empty slot) would allow the inaccessible homes to remain untouched.

Additionally, the economy config does not seem to support setting costs for particular homes. It would be great to be able to have varying costs for /home 1, /home 2 etc.

**Comment by mdcfe** [comment, marked as answer]
> After elevating permissions to allow for more homes to be set, a player has continued access to that number of homes even when removing the permission (bug or feature?). It would be ideal to have a config option to hide the extra homes until the permission is restored.

This is intentional and is unlikely to change; the multiple sethome cap applies *specifically* to `/sethome`. In addition, homes can be set for users by admins that can exceed the home limit, and applying the sethome limit to accessing homes would break this ability entirely.

Additionally, EssentialsX has never stored a numerical index of homes, and arbitrarily adding these to existing setups would likely be more confusing than helpful.

If you're concerned about players exceeding their sethome limit, you should avoid creating scenarios where players' limits will be lowered after they've been raised.

**Comment by accLarson** [comment]
Would you please reconsider in another light with a better implementation?

I would like to see a config option `enforce-home-limit` default false.
setting this to true would not permit a player to use any home related commands if their home count is higher than what is assigned to them via `sethome-multiple` permissions.

> "You have more homes than you are allowed (x). Please delete (y) homes."

**Reply by accLarson** [reply]
@mdcfe hopefully this ping is alright. Please consider my comment above.

**Reply by mdcfe** [reply]
I'd be open to a PR to add this, but I still believe this is a tiny use case that would be better served by a dedicated homes plugin.
